gitea/auth.go

```diff
@@ -0,0 +1,69 @@
+/*
```

Comment:
Same comment that I left on https://github.com/fluxcd/go-git-providers/pull/216/files#r1213407207 applies here, too:

> Why is this func in a file named auth.go? Wouldn't something like client.go be more appropriate?

Reply:
that's historic. The first 2 implementation use that naming... we keep following it. And a `client.go` does already exist.

Reply:
Sure, we can do that.